test/files/neg/t8323.scala

```diff
@@ -0,0 +1,7 @@
+object Test {
+  final val a = "Bippy"
+  final val b = "Dingo"
+  def f(x: a.type) = x
+  def f(x: b.type) = x
+  def main(args: Array[String]): Unit = ()
+}
```

Comment:
It's possible now if the result types differ sufficiently to distinguish the overloads for the JVM post-erasure.